extensions/interactions/NumericInput/directives/oppia-short-response-numeric-input.component.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the NumericInput short response.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+
+@Component({
+  selector: 'oppia-short-response-numeric-input',
+  templateUrl: './numeric-input-short-response.component.html',
+  styleUrls: []
+})
+export class ShortResponseNumericInput implements OnInit {
+  @Input() answer: string;
+  displayAnswer: Object;
+
+  constructor(
+    private htmlEscaperService: HtmlEscaperService
+  ) {}
+
+  ngOnInit(): void {
+    this.displayAnswer = this.htmlEscaperService.escapedJsonToObj(this.answer);
+    if ((this.displayAnswer as number) % 1 === 0) {
```

Comment:
This will always be true right?

Reply:
Yes, this is here for checking if displayAnswer is number or not.

Reply:
https://github.com/oppia/oppia/pull/12753/files/90765fba2c42e9dc5ff175fe132581c8d76ee1b2#diff-5c2f8ef5da16876ef9715d97bad365558e37c2b7339e9dc3d106c1f2d483e327L38-L40. Since this was check was present in the old code, I think we can keep it for now.